sidecar.sh

```diff
@@ -43,35 +42,56 @@ populate_notebook() {
 function populate() {
     log "querying manifest service at $GEN3_ENDPOINT/manifests/"
     MANIFESTS=$(curl -s -H "Authorization: Bearer ${ACCESS_TOKEN}" "https://$GEN3_ENDPOINT/manifests/")
-    while [ -z "$MANIFESTS" ]; do
-        log "Unable to get manifests from '$GEN3_ENDPOINT/manifests/'"
-        log $MANIFESTS
+    log "querying manifest service at $GEN3_ENDPOINT/metadata/"
+    METADATA=$(curl -s -H "Authorization: Bearer ${ACCESS_TOKEN}" "https://$GEN3_ENDPOINT/metadata/")
+
+    while [ -z "$MANIFESTS" ] || [ -z "$METADATA" ]; do
+        if [ -z "$MANIFESTS" ]; then
+            log "Unable to get manifests from '$GEN3_ENDPOINT/manifests/'"
+            log $MANIFESTS
+        fi
+        if [ -z "$METADATA" ]; then
+            log "Unable to get metadata from '$GEN3_ENDPOINT/metadata/'"
+            log $METADATA
+        fi
         log "sleeping for 15 seconds before trying again.."
         sleep 15
         MANIFESTS=$(curl -s -H "Authorization: Bearer ${ACCESS_TOKEN}" "https://$GEN3_ENDPOINT/manifests/")
+        METADATA=$(curl -s -H "Authorization: Bearer ${ACCESS_TOKEN}" "https://$GEN3_ENDPOINT/metadata/")
     done
+    log "successfully retrieved manifests and metadata for user"
 
+    process_files() {
+        local data=$1
+        local base_dir=$2
 
-    #  Loop over each exported manifest
-    echo $MANIFESTS | jq -c '.manifests[]' | while read i; do
+        echo $data | jq -c '.[]' | while read i; do
         FILENAME=$(echo "${i}" | jq -r .filename)
         FOLDERNAME=$(echo "${FILENAME%.*}")
-        FOLDER="/data/${GEN3_ENDPOINT}/exported-${FOLDERNAME}"
+        FOLDER="/data/${GEN3_ENDPOINT}/exported-${base_dir}/exported-${FOLDERNAME}"
 
         if [ ! -d "$FOLDER" ]; then
             log "mkdir -p $FOLDER"
             mkdir -p $FOLDER
 
             # make sure folder can be written to by notebook
             chown -R 1000:100 $FOLDER
-            MANIFEST=$(curl -s -H "Authorization: Bearer ${ACCESS_TOKEN}" "https://$GEN3_ENDPOINT/manifests/file/$FILENAME")
-            echo "${MANIFEST}" > $FOLDER/manifest.json
 
-            log "Creating notebook for $FILENAME"
-            cp ./template_manifest.json $FOLDER/data.ipynb
-            populate_notebook "$MANIFEST" "$FOLDER"
+            if ["$base_dir" == "manifests"];then
+                MANIFEST=$(curl -s -H "Authorization: Bearer ${ACCESS_TOKEN}" "https://$GEN3_ENDPOINT/manifests/file/$FILENAME")
```

Comment:
nitpick: probably better to rename this to `MANIFEST_FILE` to avoid confusions

Reply:
Agreed, fixed

Reply:
Sorry I should be more clear: I think we should keep the name the variable that holds the return value from `/manifests` unchanged as `MANIFEST`, but to update the name of the variable that holds the return value from `/manifests/file/<FILE_ID>` to `MANIFEST_FILE`, to better diff them

Reply:
and changes to `METADATA` and `METADATA_FILE` in the same way